fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleModelRepaymentPeriod.java

```diff
@@ -154,7 +154,7 @@ public boolean isRecalculatedInterestComponent() {
     @Override
     public void addInterestAmount(Money interestDue) {
         this.interestDue = this.interestDue.plus(interestDue);
-        this.totalDue = this.totalDue.plus(principalDue);
+        this.totalDue = this.totalDue.plus(interestDue);
```

Comment:
Looks like a different bug got squashed here. What effects could this bug have been producing? Shall we create a different ticket to track this?

Reply:
yes, found this while testing the new feature.

Reply:
In the add Interest method principle was added to the total due instead of the interest. which is wrong, as the added interest must reflect in the total due.